bim_webviewer/bim_web_viewer.md

```diff
@@ -1727,6 +1727,94 @@ Model
 
 ---
 
+### Set Object Color
+
+<p class="heading-link-container"><a class="heading-link" href="#set-object-color"></a></p>
+
+```js
+setObjectColor(palettes);
+```
+
+#### Description
+
+Changes the color of an object while in three modes; `default`, `selected` or `xray`. Default is while the object is neither selected or in X Ray mode. Selected is while the object is selected. X Ray is while in X Ray mode.
+
+#### Parameters
+
+| Field Name | Required | Type | Description |
+| - | - | - | - |
+| palettes | true | Object[] | Color palettes with object ids: {palette: {default: {color: '#ff0000', opacity: 1}, xray: {color: '#00ff00', opacity: 1}, selected: {color: '#0000ff', opacity: 1}}, objectIds: [1, 2, 3]}. Must provide at least one of `default`, `xray` or `selected` color. If the `color` property is not provided, then default colors are used. |
```

Comment:
Bit unreadable eh?

![image](https://user-images.githubusercontent.com/2865469/232570178-9b9cb6c6-74fb-4884-b8b6-16955a497d0d.png)

This is not the only case that has this issue, but it does seem like it'll be the worst of them.

Maybe we could break that part out into another section? Here's a suggestion. Note I also changed the usage of "palette" to "colorDescriptor" because it was being used for both the whole object and also a field of the object. Seemed like we needed another word.

Also I can't really do a codeblock here because there are codeblocks in the suggestion so begin example here:

---
```js
setObjectColor(colorDescriptors);
```

#### Description

Changes the color of an object while in three modes; `default`, `selected` or `xray`. Default is while the object is neither selected or in X Ray mode. Selected is while the object is selected. X Ray is while in X Ray mode.

#### Parameters

| Field Name | Required | Type | Description |
| - | - | - | - |
| colorDescriptor | true | ColorDescriptor[] | See `ColorDescriptor` |

##### `ColorDescriptor`

```
<!-- this part not in a codeblock, but it needs to be here for the github comment to not render it -->
<p class="heading-link-container"><a class="heading-link" href="#colordescriptor"></a></p>
```

A `ColorDescriptor` is composed of a palette, which may contain at least one of `default`, `xray`, or `selected`, and an array of object ids to apply it to.

```js
{
  palette: {
    default: {color: '#ff0000', opacity: 1},
    xray: {color: '#00ff00', opacity: 1},
    selected: {color: '#0000ff', opacity: 1}
  },
  objectIds: [1, 2, 3]
}
```

Reply:
We can also include the JSON schema, since that is what we validate against.